services/slam/slam.go

```diff
@@ -94,6 +94,22 @@ func FromDependencies(deps resource.Dependencies, name string) (Service, error)
 }
 
 // Service describes the functions that are available to the service.
+// 
+// Position example:
+//     // Get the current position of the specified source component
+//     // in the SLAM map as a Pose.
+//     pos, name, err := mySLAMService.Position(context.Background())
+// PointCloudMap example:
+//     // Get the point cloud map in standard PCD format.
+//     pcd_map, err := mySLAMService.PointCloudMap(
+//         context.Background(), true)
+// InternalState example:
+//     // Get the internal state of the SLAM algorithm required
+//     // to continue mapping/localization.
+//     internalState, err := mySlam.InternalState(context.Background())
```

Comment:
so these apis are streaming apis that return callback functions to get the chunks of the pointcloud(rather than the full pointcloud)

if you want to have these examples return the full pointcloud/internal state, then we should change them to use our helper functions.

`pcdBytes, err := PointCloudMapFull(context.Background(), mySLAMService, true)
`

`internalStateBytes, err := InternalStateFull(context.Background(), mySLAMService)`

you can probably remove the `Bytes` term. Also if you wanted to return a pointcloud.PointCloud instead of bytes lmk and I can dig up that example. its only one additional step

Reply:
@skyleilani I'm following your lead here since I think you've spent more time on this. Do we want to change the docs-side one as well as this one?

Reply:
Changed and added a note!